ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -912,3 +917,29 @@ func checkActiveDeadlineAndUpdateStatusIfNeeded(ctx context.Context, rayJob *ray
 	rayJob.Status.Message = fmt.Sprintf("The RayJob has passed the activeDeadlineSeconds. StartTime: %v. ActiveDeadlineSeconds: %d", rayJob.Status.StartTime, *rayJob.Spec.ActiveDeadlineSeconds)
 	return true
 }
+
+func checkTransitionGracePeriodAndUpdateStatusIfNeeded(ctx context.Context, rayJob *rayv1.RayJob) bool {
+	logger := ctrl.LoggerFrom(ctx)
+	if rayv1.IsJobTerminal(rayJob.Status.JobStatus) && rayJob.Status.JobDeploymentStatus == rayv1.JobDeploymentStatusRunning {
+		submitterGracePeriodTime, err := strconv.Atoi(os.Getenv(utils.RAYJOB_STATUS_TRANSITION_GRACE_PERIOD_SECONDS))
+		if err != nil {
+			submitterGracePeriodTime = utils.DEFAULT_RAYJOB_STATUS_TRANSITION_GRACE_PERIOD_SECONDS
+		}
+
+		if time.Now().Before(rayJob.Status.EndTime.Add(time.Duration(submitterGracePeriodTime) * time.Second)) {
+			return false
+		}
+		logger.Info("The RayJob has passed the submitter complete grace period. Transition the status to `Failed` or `Complete`.", "StartTime", rayJob.Status.StartTime, "submitterGracePeriodTime", submitterGracePeriodTime)
+		if rayJob.Status.JobStatus == rayv1.JobStatusFailed {
```

Comment:
nit

```go
rayJob.Status.JobDeploymentStatus = rayv1.JobDeploymentStatusComplete
if rayJob.Status.JobStatus == rayv1.JobStatusFailed {
        rayJob.Status.JobDeploymentStatus = rayv1.JobDeploymentStatusFailed
}
```

Reply:
keep the logic similar to https://github.com/ray-project/kuberay/blob/bdfb012125c74f991bdfc06f30b9b02aa9a5cc70/ray-operator/controllers/ray/rayjob_controller.go#L296-L300

Reply:
maybe you can add a util function to unify the logic:

```
func xxxxx(jobStatus) JobDeploymentStatus {
  ...
}
```